bagua/torch_api/data_parallel/distributed.py

```diff
@@ -0,0 +1,350 @@
+# pytype: disable=module-attr
+import torch
+from torch.nn.modules import Module
+from torch._C._distributed_c10d import ProcessGroup as TorchProcessGroup
+from torch.nn.parallel import DistributedDataParallel as TorchDistributedDataParallel
+from typing import Callable, List, Optional, Union
+import warnings
+
+import bagua
+from bagua.torch_api.algorithms.gradient_allreduce import GradientAllReduceAlgorithm
+from contextlib import contextmanager
+from bagua.torch_api.communication import (
+    _get_default_group,
+    BaguaProcessGroup,
+)
+from .bagua_distributed import BaguaDistributedDataParallel
+
+
+class DistributedDataParallel_V1_9_0_Interface(Module):
+    r"""
+    PyTorch v1.9.0 DistributedDataParallel(https://github.com/pytorch/pytorch/blob/v1.9.0/torch/nn/parallel/distributed.py#L125) compatible interface.
+    """
+
+    def __init__(self) -> None:
+        super(DistributedDataParallel_V1_9_0_Interface, self).__init__()
+
+    @contextmanager
+    def no_sync(self):
+        raise NotImplementedError
+
+    def forward(self, *inputs, **kwargs):
+        raise NotImplementedError
+
+    def scatter(self, inputs, kwargs, device_ids):
+        raise NotImplementedError
+
+    def to_kwargs(self, inputs, kwargs, device_id):
+        raise NotImplementedError
+
+    def gather(self, outputs, output_device):
+        raise NotImplementedError
+
+    def train(self, mode=True):
+        raise NotImplementedError
+
+    @contextmanager
+    def join(
+        self,
+        divide_by_initial_world_size=True,
+        enable=True,
+        throw_on_early_termination=False,
+    ):
+        raise NotImplementedError
+
+    def register_comm_hook(self, state: object, hook: Callable):
+        raise NotImplementedError
+
+    def will_sync_module_buffers(self):
+        raise NotImplementedError
+
+
+def to_bagua_process_group(process_group: Union[TorchProcessGroup, BaguaProcessGroup, None] = None):
+    """Torch process group to bagua converter.
+
+    Args:
+        process_group (Union[TorchProcessGroup, BaguaProcessGroup, None], optional): Map
```

Comment:
Here we only need to explain the argument.

For example: PyTorch process group or Bagua process group. The default PyTorch process group is used if `None` is passed in.